joi-validations/profiles.validations.js

```diff
@@ -27,7 +27,23 @@ const ProfileValidation = {
   unfollowAnonUser: {
     body: object({
       follow_source: string.required(),
-      user_id_followed: uuid.required()
+      user_id_followed: uuid.when('follow_source', {
+        switch: [
+          {is: 'post', then: Joi.optional()},
+          {is: 'comment', then: Joi.optional()}
+        ],
+        otherwise: Joi.required()
+      }),
+      post_id: uuid.when('follow_source', {
+        is: 'post',
+        then: Joi.required(),
+        otherwise: Joi.optional()
+      }),
+      comment_id: uuid.when('follow_source', {
+        is: 'comment',
+        then: Joi.required(),
+        otherwise: Joi.optional()
+      })
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made to the validation schema in `profiles.validations.js` are logical and seem to be correctly implemented. The conditional validation based on the `follow_source` property is a good approach to handle different scenarios for unfollowing an anonymous user. However, it would be beneficial to add some error handling or assertions to ensure that the `follow_source` value is one of the expected values ("post", "comment", or any other valid value). This will prevent unexpected behavior if an invalid `follow_source` value is provided.
```diff
   unfollowAnonUser: {
     body: object({
       follow_source: string.valid('post', 'comment').required(),
      user_id_followed: uuid.when('follow_source', {
        switch: [
          {is: 'post', then: Joi.optional()},
          {is: 'comment', then: Joi.optional()}
        ],
        otherwise: Joi.required()
      }),
      post_id: uuid.when('follow_source', {
        is: 'post',
        then: Joi.required(),
        otherwise: Joi.optional()
      }),
      comment_id: uuid.when('follow_source', {
        is: 'comment',
        then: Joi.required(),
        otherwise: Joi.optional()
      })
     })
   }
 };
```
This change ensures that the `follow_source` value must be either "post" or "comment". If a different value is provided, Joi will return a validation error.


<!-- This is an auto-generated comment by OSS CodeRabbit -->